library/src/main/scala/g8.scala

```diff
@@ -462,35 +474,45 @@ object G8 {
     }
   }
 
-  def readProps(stm: java.io.InputStream): G8.OrderedProperties = {
-    val p = new LinkedListProperties
-    p.load(stm)
-    stm.close()
-    (OrderedProperties.empty /: p.keyList) { (l, k) =>
-      l :+ (k -> p.getProperty(k))
+  def readProps(stm: InputStream): G8.OrderedProperties = {
+    // Overrides java.util.Properties to return properties back in the order they were specified
+    val properties = new Properties {
+      val parameters = mutable.MutableList.empty[(String, String)]
+
+      override def put(key: Object, value: Object) = {
+        val k = key.toString
+        val v = value.toString
+
+        parameters += k -> v
+        super.put(k, v)
+      }
+    }
+
+    try {
+      properties.load(stm)
+      properties.parameters.toList
+    } finally {
+      stm.close()
     }
   }
+
+  private val truthy = Set("y", "yes", "true")
+  private val falsy = Set("n", "no", "false")
```

Comment:
@eed3si9n i understand that but, sorry, i fail to see how empty collections and iterators could be relevant in a context where all we have are strings read from a properties file. i'm missing your point here. what strategy do you propose we adopt?